pennylane/templates/state_preparations/multiplexer_state_prep.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2018-2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""Contains the MultiplexerStatePreparation template."""
+
+import numpy as np
+
+import pennylane as qml
+from pennylane.operation import Operation
+from pennylane.wires import Wires
+
+
+class MultiplexerStatePreparation(Operation):
+    r"""Prepares a quantum state using multiplexed rotations.
+
+    This operation implements the state preparation method described
+    in `arXiv:0208112 <https://arxiv.org/abs/quant-ph/0208112>`_.
+
+    Args:
+        state_vector (tensor_like): The state vector of length :math:`2^n` to be prepared on :math:`n` wires.
+        wires (Sequence[int]): The wires on which to prepare the state.
+
+    Raises:
+        ValueError: If the length of the input state vector array is not :math:`2^n` where :math:`n` is an integer, or if
+            its norm is not equal to one.
+
+    **Example**
+
+    .. code-block:: python
+
+        import numpy as np
+
+        probs_vector = np.array([0.5, 0., 0.25, 0.25])
+
+        dev = qml.device("default.qubit", wires = 2)
+
+        wires = [0,1]
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.MultiplexerStatePreparation(
+                np.sqrt(probs_vector), wires
+            )
+            return qml.probs(wires)
+
+    .. code-block:: pycon
+
+        >>> np.round(circuit(), 2)
+        array([0.5 , 0.  , 0.25, 0.25])
+
+    .. seealso:: :class:`~.SelectPauliRot`
+
+    """
+
+    resource_keys = {"num_wires"}
+
+    # pylint: disable=too-many-positional-arguments
+    def __init__(self, state_vector, wires, id=None):  # pylint: disable=too-many-arguments
+
+        n_amplitudes = qml.math.shape(state_vector)[0]
+        if n_amplitudes != 2 ** len(Wires(wires)):
+            raise ValueError(
+                f"State vectors must be of length {2 ** len(wires)}; vector has length {n_amplitudes}."
+            )
+
+        norm = qml.math.linalg.norm(state_vector)
```

Comment:
Should we always do the norm computation?

Reply:
What do you mean here? Maybe adding an attribute "check"?
(I based this in QROMStatePreparation)